README.md

```diff
@@ -1,3 +1,46 @@
+# Azure Naming
+
+This module helps you to keep consistency on your resources names for terraform The goal of this module it is that for each resource that requires a name in terraform you would be easialy able to compose this name using this module and this will keep the consistency in your repositories.
+
+# Usage
+
+For every resource in `terraform_azurerm` just remove the `azurerm` part of the module and use the `name` property of this output.
+
+example for `azurerm_resource_group` you can use :
+
+```tf
+module "naming" "main" {
+  suffix = [ "test" ]
+}
+resource "azurerm_resource_group" "example" {
+  name     = module.naming.main.resource_group.name
+  location = "West Europe"
+}
+```
+
+if you want this to be unique for this module and not shared with other instances of this module you can use `name_unique`
+
+```tf
+module "naming" "main" {
+  suffix = [ "test" ]
+}
+resource "azurerm_resource_group" "example" {
+  name     = module.naming.main.resource_group.name
+  location = "West Europe"
+}
+```
+There are other advances usages that will be explained in the [Advanced usages](#advancedusages) part og this docs.
+
+# Internals
+
+## Modifying resources
+
+The resources are automatically generated usign `go` to change the generation please change the file on the `templates` folder. To add a new resource, correct one implemented or implement one of the missing ones add the resource definition to the `resourceDefinition.jsonc` file it will be automatically generated.
```

Comment:
"To add a new resource, correct one implemented or implement one of the missing ones add the resource definition to the `resourceDefinition.jsonc` file it will be automatically generated." - this is confusing